spring-cloud-gcp-autoconfigure/src/main/java/com/google/cloud/spring/autoconfigure/sql/R2dbcDatabaseType.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright 2021-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.spring.autoconfigure.sql;
+
+public enum R2dbcDatabaseType {
```

Comment:
Would it make sense to just add another property to `DatabaseType` for R2DBC url template?

Reply:
Hm `DatabaseType` includes some extra information, particularly the driver class name that is later set through `spring.datasource.driverClassName`. A similar option doesn't seem to exist for R2DBC so I decided to separate them out on the basis of what data is required for each. Additionally, [`ConnectionFactoryOptions`](https://github.com/r2dbc/r2dbc-spi/blob/d27db1062a9e91417c8c9bc2918d0db9b9f30d26/r2dbc-spi/src/main/java/io/r2dbc/spi/ConnectionFactoryOptions.java#L53) in the R2DBC SPI includes a "Driver" option but our GcpConnectionFactoryProviders end up overriding that value ([ as seen here for example](https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/2fc784f272c74c18b54269cd1bcf8214737aa11f/r2dbc/postgres/src/main/java/com/google/cloud/sql/core/GcpConnectionFactoryProviderPostgres.java#L79)) 

However, I was wondering, was there a reason why we decided to provide the driver class name through the property instead of letting Spring boot determine it from the url?

Reply:
Why would it be a problem if `DatabaseType` contained constants for both JDBC and R2DBC?
It could have:
`jdbcUrlTemplate, r2dbcUrlTemplate, defaultUsername`

Reply:
My bad! You're right. I take back my comment.